### ISSUE TITLE
slurm mandatory `partition` and non-mandatory `qos` are not always mandatory and non-mandatory.

### ISSUE THREAD

**Issue by benclifford** [Parsl/libsubmit#66]
On `cori.nersc.gov`, the (mandatory in libsubmit) slurm `--partition` parameter is not mandatory (and documented here as causing jobs to fail, although that is not always what happens in practice: http://www.nersc.gov/users/computational-systems/cori/running-jobs/queues-and-policies/ ).

The `qos` parameter, which roughly replaces it *is* mandatory, but it is not mandatory in libsubmit.

Both of these can be specified via `overrides` (as well as other mandatory cori parameters such as `--constraint`) like this:
```
                overrides="#SBATCH --qos=debug\n#SBATCH -C knl"
```

My suggestion is that `partition` cease to become mandatory, perhaps be removed entirely, and perhaps a better user interface for presenting `SBATCH` directives be implemented (for example a list of strings without the #SBATCH prefix?)

**Comment by TomGlanzman** [comment]
Using Parsl 0.9.0 (special Ben Clifford tag for LSST-DESC DM) runs at NERSC and creates submit files that contain the "--partition" option.  As of today, this seems to work for "debug" and "regular" queues...but not for "premium".  By simply changing to "premium", one sees this complaint from Parsl:

```2019-12-12 11:55:17 parsl.providers.slurm.slurm:191 [DEBUG]  Requesting one block with 1 nodes
2019-12-12 11:55:17 parsl.providers.slurm.slurm:208 [DEBUG]  Writing submit script
2019-12-12 11:55:17 parsl.providers.slurm.slurm:212 [DEBUG]  moving files
Submission of command to scale_out failed
2019-12-12 11:55:17 parsl.providers.slurm.slurm:228 [ERROR]  Retcode:1 STDOUT: STDERR:sbatch: error: Job request does not match any supported policy for cori
sbatch: error: Batch job submission failed: Unspecified error
2019-12-12 11:55:17 parsl.executors.high_throughput.executor:566 [DEBUG]  Launched block 0->None
2019-12-12 11:55:17 parsl.dataflow.flow_control:142 [ERROR]  Flow control callback threw an exception - logging and proceeding anyway
Traceback (most recent call last):
  File "/global/homes/d/descdm/.conda/envs/parsl-lsst-dm/lib/python3.7/site-packages/parsl/dataflow/flow_control.py", line 140, in make_callback
    self.callback(tasks=self._event_buffer, kind=kind)
  File "/global/homes/d/descdm/.conda/envs/parsl-lsst-dm/lib/python3.7/site-packages/parsl/dataflow/strategy.py", line 267, in _strategy_simple
    executor.scale_out(1)
  File "/global/homes/d/descdm/.conda/envs/parsl-lsst-dm/lib/python3.7/site-packages/parsl/executors/high_throughput/executor.py", line 569, in scale_out
    "Attempts to provision nodes via provider has failed"))
parsl.executors.errors.ScalingFailed: ('slurm', 'Attempts to provision nodes via provider has failed')
```
I see a suggested work-around is to specify "-qos" in the overrides.  Still, it would be nice to see "--partition" replaced with "--qos" soon.

**Comment by TomGlanzman** [comment]
As a post script, I also observe that in the condition described above, Parsl will continue to submit jobs to SLURM indefinitely.  If this case (or similar) can be easily identified, it might be good to abort the workflow.

**Comment by benclifford** [comment]
fixed in parsl #1501
